cpp/include/raft/random/rng.hpp

```diff
@@ -0,0 +1,376 @@
+/*
```

Comment:
cugraph uses rng.cuh in a couple of places, so leaving a quick review so we don’t merge this before that side is ready (if it already is, just let me know and glad to approve/merge)